src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -167,5 +168,7 @@ private static string CreateExtraPropertyDiscriminatorSummary(CSharpType valueTy
             }
             return updatedDescription;
         }
+
+        public bool ShouldSkipDeserialization => InputModelProperty?.Type is InputLiteralType;
```

Comment:
I think we can move this into `ModelTypeProvider` and call inside `CreatePropertySerializations`. I wonder maybe we should have two sets of serialization properties - for serialization and for deserialization.

Reply:
Previously I implemented this property in `ObjectTypeProperty` because there are other 2 places (`SerializationBuilder.GetPropertySerializationsFromBag` and `JsonAddtiontalPropertiesSerialization`) which handle `JsonPropertySerliaization`. I rechecked and confirmed that we won't skip deserialization in those 2 places. So I move this check into `ModelTypeProvider`.

I think currently `JsonPropertySerialization` should be sufficient to represent the serialization constraints.